src/api/api.ts

```diff
@@ -0,0 +1,36 @@
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { client } from '../utils/fetchClient';
+
+export const getUsers = () => {
+  return client.get<User[]>('/users');
+};
+
+export const getPosts = ({ id }: Pick<User, 'id'>) => {
+  return client.get<Post[]>(`/posts?userId=${id}`);
```

Comment:
Similar to `getUsers`, add error handling for the `getPosts` function to handle any issues that may arise during the API call.